tests/behat/features/migrate.mappings.components.references.feature

```diff
@@ -0,0 +1,110 @@
+    And I clear "node_civictheme_page_annotate" migration map
+    And I run drush "config-set migrate_plus.migration.media_civictheme_image source.urls []"
+    And I run drush "config-set migrate_plus.migration.media_civictheme_document source.urls []"
+    And I run drush "config-set migrate_plus.migration.node_civictheme_page source.urls []"
+    And I run drush "config-set migrate_plus.migration.node_civictheme_page_annotate source.urls []"
+
+  @api @drush
+  Scenario: Migration local sources can be updated from the migration edit form
+    Given I am logged in as a user with the "administrator" role
+
+    # Attaching 2 source image files as the node migration depends on images in both files.
+    When I go to "admin/structure/migrate/manage/civictheme_migrate/migrations/media_civictheme_image/edit"
+    And I attach the file "migrate/civictheme_migrate.media_civictheme_image_1.json" to "files[source_update_files][]"
+    And I press "Update Migration"
+    When I go to "admin/structure/migrate/manage/civictheme_migrate/migrations/media_civictheme_image/edit"
+    And I attach the file "migrate/civictheme_migrate.media_civictheme_image_2.json" to "files[source_update_files][]"
+    And I press "Update Migration"
+
+    When I go to "admin/structure/migrate/manage/civictheme_migrate/migrations/media_civictheme_document/edit"
+    And I attach the file "migrate/civictheme_migrate.media_civictheme_document_1.json" to "files[source_update_files][]"
+    And I press "Update Migration"
+
+    When I go to "admin/structure/migrate/manage/civictheme_migrate/migrations/node_civictheme_page/edit"
+    And I attach the file "migrate/civictheme_migrate.node_civictheme_page_7.json" to "files[source_update_files][]"
+    And I press "Update Migration"
+
+    When I go to "admin/structure/migrate/manage/civictheme_migrate/migrations/node_civictheme_page_annotate/edit"
+    And I attach the file "migrate/civictheme_migrate.node_civictheme_page_7.json" to "files[source_update_files][]"
+    And I press "Update Migration"
+
+    When I run drush "mim --group=civictheme_migrate"
+    And I visit "civictheme_page" "[TEST] Migrated Page Content 71"
+
+    # Asserting field mappings.
+    #Alias
+    Then I should be in the "/migrated/page-content-71" path
+    #Banner
+    Then I should see "[TEST] Banner title - Migrated Page Content 71" in the ".ct-banner__title" element
+    #Content
+    And I should see an ".ct-basic-content" element
+
+    And I visit "civictheme_page" "[TEST] Migrated Page Content 72"
+    # Asserting field mappings.
+    #Alias
+    Then I should be in the "/migrated/page-content-72" path
+    #Banner
+    Then I should see "[TEST] Banner title - Migrated Page Content 72" in the ".ct-banner__title" element
+    #Content
+    And I should see an ".ct-basic-content" element
+    And I should see the text "[TEST] Basic text content"
+    #attachment
+    And I should see an ".ct-figure" element
```

Comment:
this test does not test for the correct references conversion

The test should check:

1. Links conversion
 a. links to existing referenced content should be converted to Linkit media items. To check that this has happened, go to page edit screen (non-js test) and check the source contains linkit media tag
 b. Links to non-existing referenced content should stay as is (relative or absolute links). Assert in the same way as above.
2. Embedded images
  a. Links to existing images should be asserted for.
  b. Links to non-existing images should be asserted for AND the presence of the created images.
  c. Referencing of the newly created images should not create duplicates (image referenced and created as a part of migration of node 1 should not be duplicated when referenced from node 2)
3. Embedded documents - same as for images

Reply:
Point 1 Test Added